specification/oracle/models/oraclesubscription/oracleSubscription.tsp

```diff
@@ -0,0 +1,177 @@
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using Azure.ResourceManager;
+
+namespace Oracle.Database;
+
+@doc("OracleSubscriptionProvisioningState enum")
+union OracleSubscriptionProvisioningState {
+  ResourceProvisioningState,
+  string,
+}
+
+@singleton
+@subscriptionResource
+@doc("OracleSubscription resource definition")
+@friendlyName("OracleSubscription")
+model OracleSubscription<ResourcePlanType>
+  is ProxyResource<OracleSubscriptionProperties> {
+  @doc("OracleSubscription name")
+  @key("oraclesubscriptionname")
+  @path
+  @segment("oracleSubscriptions")
+  @pattern(".*")
+  name: string;
+
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-invalid-envelope-property"
+  @doc("Details of the resource plan.")
+  plan?: ResourcePlanType;
+}
+
+// Azure specs do not support update model for resource plan
+@doc("ResourcePlanTypeUpdate model definition")
+model PlanUpdate {
+  @doc("A user defined name of the 3rd Party Artifact that is being procured.")
+  name?: string;
+
+  @doc("The publisher of the 3rd Party Artifact that is being bought. E.g. NewRelic")
+  publisher?: string;
+
+  @doc("The 3rd Party artifact that is being procured. E.g. NewRelic. Product maps to the OfferID specified for the artifact at the time of Data Market onboarding. ")
+  product?: string;
+
+  @doc("A publisher provided promotion code as provisioned in Data Market for the said product/artifact.")
+  promotionCode?: string;
+
+  @doc("The version of the desired product/artifact.")
+  version?: string;
+}
+
+@doc("CloudAccountProvisioningState enum")
+union CloudAccountProvisioningState {
+  @doc("Pending - Initial state when Oracle cloud account is not configured")
+  Pending: "Pending",
+
+  @doc("Provisioning - State when Oracle cloud account is being provisioned")
+  Provisioning: "Provisioning",
+
+  @doc("Available - State when Oracle cloud account cloud linking is complete and it is available")
+  Available: "Available",
+
+  string,
+}
+
+@doc("Intent enum")
+union Intent {
+  @doc("Retain intent")
+  Retain: "Retain",
+
+  @doc("Reset intent")
+  Reset: "Reset",
+
+  string,
+}
+
+@doc("Cloud Account Details model")
+model CloudAccountDetails {
+  @doc("Cloud Account name")
+  @visibility("read")
+  cloudAccountName?: string;
+
+  @doc("Cloud Account Home region")
+  @visibility("read")
+  cloudAccountHomeRegion?: string;
+}
+
+@doc("SaaS Subscription Details model")
+model SaasSubscriptionDetails {
+  @doc("Purchased SaaS subscription ID")
+  @visibility("read")
+  id?: string;
+
+  @doc("SaaS subscription name")
+  @visibility("read")
+  subscriptionName?: string;
```

Comment:
A little curious, are you sure the visibility is being set correctly? subscriptionName should be able to be set by customers?  so as some other properties

Reply:
We are not allowing customer to set subscriptionName, because it is set by default for singleton resource and not supposed to be changed